src/common.cpp

```diff
@@ -1993,3 +1994,26 @@ void redirect_tty_output() {
     show_stackframe(L'E', 99, 1);
     abort();
 }
+
+/// Return a path to a directory where we can store temporary files.
+wcstring get_path_to_tmp_dir()
+{
+    env_var_t env_tmpdir = env_get_string(L"TMPDIR");
+    if (!env_tmpdir.missing()) return env_tmpdir;
+
+#ifdef _CS_DARWIN_USER_TEMP_DIR
+    char osx_tmpdir[31];
```

Comment:
Why 31 bytes? In general magic values like this should be avoided or at least given a symbolic name via `#define`. In this case you should use `PATH_MAX` (which has its own problems but is better than a hardcoded value).